mapUsingPython: explicitly check availability of Python, produce a user-friendly message

When running in a Docker image that doesn't include Python, `PythonTransforms.mapUsingPythonBatch` unsurprisingly fails. 

However, the failure cause is not shown, which makes it difficult for the user to diagnose, you only get

```
Caused by: java.lang.Exception: Initialization script finished with non-zero exit code: 127
```

So this is user error ultimately, but little diagnostic support.

Steps to recreate, create a Docker image using `FROM library/openjdk:11-jre-slim` that starts a Jet server, and run that Jet server. Then inject the job from a Jet client.

I get
```
11:53:53.940 [hz.competent_lalande.cached.thread-3] WARN  com.hazelcast.jet.impl.processor.ProcessorSupplierWithService.mapUsingPythonBatch#ProcessorSupplier - PythonService context creation failed,  will retry
 com.hazelcast.jet.JetException: PythonService initialization failed: java.lang.Exception: Initialization script finished with non-zero exit code: 127
	at com.hazelcast.jet.python.PythonServiceContext.<init>(PythonServiceContext.java:110) ~[hazelcast-jet-python-4.4-SNAPSHOT.jar!/:4.4-SNAPSHOT]
	at com.hazelcast.jet.python.PythonService.createContextWithRetry(PythonService.java:111) ~[hazelcast-jet-python-4.4-SNAPSHOT.jar!/:4.4-SNAPSHOT]
	at com.hazelcast.jet.python.PythonService.lambda$factory$9a0f8f23$1(PythonService.java:91) ~[hazelcast-jet-python-4.4-SNAPSHOT.jar!/:4.4-SNAPSHOT]
	at com.hazelcast.function.FunctionEx.apply(FunctionEx.java:47) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.processor.ProcessorSupplierWithService.init(ProcessorSupplierWithService.java:55) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.execution.init.ExecutionPlan.initProcSuppliers(ExecutionPlan.java:327) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.execution.init.ExecutionPlan.initialize(ExecutionPlan.java:158) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.execution.ExecutionContext.initialize(ExecutionContext.java:138) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.JobExecutionService.lambda$initExecution$808bb183$1(JobExecutionService.java:240) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.function.RunnableEx.run(RunnableEx.java:30) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.util.Util.doWithClassLoader(Util.java:446) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.JobExecutionService.initExecution(JobExecutionService.java:240) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.operation.InitExecutionOperation.run(InitExecutionOperation.java:73) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.spi.impl.operationservice.Operation.call(Operation.java:184) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.call(OperationRunnerImpl.java:227) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:216) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationExecutorImpl.run(OperationExecutorImpl.java:406) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationExecutorImpl.runOrExecute(OperationExecutorImpl.java:433) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.doInvokeLocal(Invocation.java:596) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.doInvoke(Invocation.java:581) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.invoke0(Invocation.java:540) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.invoke(Invocation.java:237) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.spi.impl.operationservice.impl.InvocationBuilderImpl.invoke(InvocationBuilderImpl.java:59) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.MasterContext.invokeOnParticipant(MasterContext.java:261) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.MasterContext.invokeOnParticipants(MasterContext.java:245) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.MasterJobContext.lambda$tryStartJob$2(MasterJobContext.java:227) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.JobCoordinationService.lambda$submitToCoordinatorThread$46(JobCoordinationService.java:1024) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.JobCoordinationService.submitToCoordinatorThread(JobCoordinationService.java:1033) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.JobCoordinationService.submitToCoordinatorThread(JobCoordinationService.java:1023) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.MasterJobContext.tryStartJob(MasterJobContext.java:189) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.JobCoordinationService.tryStartJob(JobCoordinationService.java:933) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.JobCoordinationService.lambda$submitJob$1(JobCoordinationService.java:258) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.JobCoordinationService.lambda$submitToCoordinatorThread$46(JobCoordinationService.java:1024) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.JobCoordinationService.lambda$submitToCoordinatorThread$47(JobCoordinationService.java:1045) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.internal.util.executor.CompletableFutureTask.run(CompletableFutureTask.java:64) [hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.internal.util.executor.CachedExecutorServiceDelegate$Worker.run(CachedExecutorServiceDelegate.java:217) [hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source) [?:?]
	at java.lang.Thread.run(Unknown Source) [?:?]
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.executeRun(HazelcastManagedThread.java:64) [hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:80) [hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
Caused by: java.lang.Exception: Initialization script finished with non-zero exit code: 127
	at com.hazelcast.jet.python.PythonServiceContext.<init>(PythonServiceContext.java:101) ~[hazelcast-jet-python-4.4-SNAPSHOT.jar!/:4.4-SNAPSHOT]
	... 40 more
11:53:54.968 [hz.competent_lalande.cached.thread-3] WARN  com.hazelcast.jet.impl.processor.ProcessorSupplierWithService.mapUsingPythonBatch#ProcessorSupplier - PythonService context creation failed,  will retry
 com.hazelcast.jet.JetException: PythonService initialization failed: java.lang.Exception: Initialization script finished with non-zero exit code: 127
	at com.hazelcast.jet.python.PythonServiceContext.<init>(PythonServiceContext.java:110) ~[hazelcast-jet-python-4.4-SNAPSHOT.jar!/:4.4-SNAPSHOT]
	at com.hazelcast.jet.python.PythonService.createContextWithRetry(PythonService.java:111) ~[hazelcast-jet-python-4.4-SNAPSHOT.jar!/:4.4-SNAPSHOT]
	at com.hazelcast.jet.python.PythonService.lambda$factory$9a0f8f23$1(PythonService.java:91) ~[hazelcast-jet-python-4.4-SNAPSHOT.jar!/:4.4-SNAPSHOT]
	at com.hazelcast.function.FunctionEx.apply(FunctionEx.java:47) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.processor.ProcessorSupplierWithService.init(ProcessorSupplierWithService.java:55) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.execution.init.ExecutionPlan.initProcSuppliers(ExecutionPlan.java:327) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.execution.init.ExecutionPlan.initialize(ExecutionPlan.java:158) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.execution.ExecutionContext.initialize(ExecutionContext.java:138) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.JobExecutionService.lambda$initExecution$808bb183$1(JobExecutionService.java:240) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.function.RunnableEx.run(RunnableEx.java:30) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.util.Util.doWithClassLoader(Util.java:446) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.JobExecutionService.initExecution(JobExecutionService.java:240) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.operation.InitExecutionOperation.run(InitExecutionOperation.java:73) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.spi.impl.operationservice.Operation.call(Operation.java:184) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.call(OperationRunnerImpl.java:227) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:216) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationExecutorImpl.run(OperationExecutorImpl.java:406) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationExecutorImpl.runOrExecute(OperationExecutorImpl.java:433) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.doInvokeLocal(Invocation.java:596) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.doInvoke(Invocation.java:581) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.invoke0(Invocation.java:540) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.invoke(Invocation.java:237) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.spi.impl.operationservice.impl.InvocationBuilderImpl.invoke(InvocationBuilderImpl.java:59) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.MasterContext.invokeOnParticipant(MasterContext.java:261) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.MasterContext.invokeOnParticipants(MasterContext.java:245) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.MasterJobContext.lambda$tryStartJob$2(MasterJobContext.java:227) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.JobCoordinationService.lambda$submitToCoordinatorThread$46(JobCoordinationService.java:1024) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.JobCoordinationService.submitToCoordinatorThread(JobCoordinationService.java:1033) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.JobCoordinationService.submitToCoordinatorThread(JobCoordinationService.java:1023) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.MasterJobContext.tryStartJob(MasterJobContext.java:189) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.JobCoordinationService.tryStartJob(JobCoordinationService.java:933) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.JobCoordinationService.lambda$submitJob$1(JobCoordinationService.java:258) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.JobCoordinationService.lambda$submitToCoordinatorThread$46(JobCoordinationService.java:1024) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.JobCoordinationService.lambda$submitToCoordinatorThread$47(JobCoordinationService.java:1045) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.internal.util.executor.CompletableFutureTask.run(CompletableFutureTask.java:64) [hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.internal.util.executor.CachedExecutorServiceDelegate$Worker.run(CachedExecutorServiceDelegate.java:217) [hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source) [?:?]
	at java.lang.Thread.run(Unknown Source) [?:?]
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.executeRun(HazelcastManagedThread.java:64) [hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:80) [hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
Caused by: java.lang.Exception: Initialization script finished with non-zero exit code: 127
	at com.hazelcast.jet.python.PythonServiceContext.<init>(PythonServiceContext.java:101) ~[hazelcast-jet-python-4.4-SNAPSHOT.jar!/:4.4-SNAPSHOT]
	... 40 more
11:53:55.978 [hz.competent_lalande.cached.thread-3] WARN  com.hazelcast.jet.impl.processor.ProcessorSupplierWithService.mapUsingPythonBatch#ProcessorSupplier - PythonService context creation failed,  giving up
 com.hazelcast.jet.JetException: PythonService initialization failed: java.lang.Exception: Initialization script finished with non-zero exit code: 127
	at com.hazelcast.jet.python.PythonServiceContext.<init>(PythonServiceContext.java:110) ~[hazelcast-jet-python-4.4-SNAPSHOT.jar!/:4.4-SNAPSHOT]
	at com.hazelcast.jet.python.PythonService.createContextWithRetry(PythonService.java:111) ~[hazelcast-jet-python-4.4-SNAPSHOT.jar!/:4.4-SNAPSHOT]
	at com.hazelcast.jet.python.PythonService.lambda$factory$9a0f8f23$1(PythonService.java:91) ~[hazelcast-jet-python-4.4-SNAPSHOT.jar!/:4.4-SNAPSHOT]
	at com.hazelcast.function.FunctionEx.apply(FunctionEx.java:47) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.processor.ProcessorSupplierWithService.init(ProcessorSupplierWithService.java:55) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.execution.init.ExecutionPlan.initProcSuppliers(ExecutionPlan.java:327) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.execution.init.ExecutionPlan.initialize(ExecutionPlan.java:158) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.execution.ExecutionContext.initialize(ExecutionContext.java:138) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.JobExecutionService.lambda$initExecution$808bb183$1(JobExecutionService.java:240) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.function.RunnableEx.run(RunnableEx.java:30) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.util.Util.doWithClassLoader(Util.java:446) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.JobExecutionService.initExecution(JobExecutionService.java:240) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.operation.InitExecutionOperation.run(InitExecutionOperation.java:73) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.spi.impl.operationservice.Operation.call(Operation.java:184) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.call(OperationRunnerImpl.java:227) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:216) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationExecutorImpl.run(OperationExecutorImpl.java:406) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationExecutorImpl.runOrExecute(OperationExecutorImpl.java:433) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.doInvokeLocal(Invocation.java:596) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.doInvoke(Invocation.java:581) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.invoke0(Invocation.java:540) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.invoke(Invocation.java:237) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.spi.impl.operationservice.impl.InvocationBuilderImpl.invoke(InvocationBuilderImpl.java:59) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.MasterContext.invokeOnParticipant(MasterContext.java:261) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.MasterContext.invokeOnParticipants(MasterContext.java:245) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.MasterJobContext.lambda$tryStartJob$2(MasterJobContext.java:227) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.JobCoordinationService.lambda$submitToCoordinatorThread$46(JobCoordinationService.java:1024) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.JobCoordinationService.submitToCoordinatorThread(JobCoordinationService.java:1033) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.JobCoordinationService.submitToCoordinatorThread(JobCoordinationService.java:1023) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.MasterJobContext.tryStartJob(MasterJobContext.java:189) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.JobCoordinationService.tryStartJob(JobCoordinationService.java:933) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.JobCoordinationService.lambda$submitJob$1(JobCoordinationService.java:258) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.JobCoordinationService.lambda$submitToCoordinatorThread$46(JobCoordinationService.java:1024) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.JobCoordinationService.lambda$submitToCoordinatorThread$47(JobCoordinationService.java:1045) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.internal.util.executor.CompletableFutureTask.run(CompletableFutureTask.java:64) [hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.internal.util.executor.CachedExecutorServiceDelegate$Worker.run(CachedExecutorServiceDelegate.java:217) [hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source) [?:?]
	at java.lang.Thread.run(Unknown Source) [?:?]
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.executeRun(HazelcastManagedThread.java:64) [hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:80) [hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
Caused by: java.lang.Exception: Initialization script finished with non-zero exit code: 127
	at com.hazelcast.jet.python.PythonServiceContext.<init>(PythonServiceContext.java:101) ~[hazelcast-jet-python-4.4-SNAPSHOT.jar!/:4.4-SNAPSHOT]
	... 40 more
11:53:56.981 [hz.competent_lalande.cached.thread-3] ERROR com.hazelcast.jet.impl.operation.InitExecutionOperation - PythonService initialization failed: java.lang.Exception: Initialization script finished with non-zero exit code: 127
 com.hazelcast.jet.JetException: PythonService initialization failed: java.lang.Exception: Initialization script finished with non-zero exit code: 127
	at com.hazelcast.jet.python.PythonServiceContext.<init>(PythonServiceContext.java:110) ~[hazelcast-jet-python-4.4-SNAPSHOT.jar!/:4.4-SNAPSHOT]
	at com.hazelcast.jet.python.PythonService.createContextWithRetry(PythonService.java:111) ~[hazelcast-jet-python-4.4-SNAPSHOT.jar!/:4.4-SNAPSHOT]
	at com.hazelcast.jet.python.PythonService.lambda$factory$9a0f8f23$1(PythonService.java:91) ~[hazelcast-jet-python-4.4-SNAPSHOT.jar!/:4.4-SNAPSHOT]
	at com.hazelcast.function.FunctionEx.apply(FunctionEx.java:47) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.processor.ProcessorSupplierWithService.init(ProcessorSupplierWithService.java:55) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.execution.init.ExecutionPlan.initProcSuppliers(ExecutionPlan.java:327) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.execution.init.ExecutionPlan.initialize(ExecutionPlan.java:158) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.execution.ExecutionContext.initialize(ExecutionContext.java:138) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.JobExecutionService.lambda$initExecution$808bb183$1(JobExecutionService.java:240) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.function.RunnableEx.run(RunnableEx.java:30) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.util.Util.doWithClassLoader(Util.java:446) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.JobExecutionService.initExecution(JobExecutionService.java:240) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.operation.InitExecutionOperation.run(InitExecutionOperation.java:73) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.spi.impl.operationservice.Operation.call(Operation.java:184) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.call(OperationRunnerImpl.java:227) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:216) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationExecutorImpl.run(OperationExecutorImpl.java:406) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationExecutorImpl.runOrExecute(OperationExecutorImpl.java:433) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.doInvokeLocal(Invocation.java:596) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.doInvoke(Invocation.java:581) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.invoke0(Invocation.java:540) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.invoke(Invocation.java:237) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.spi.impl.operationservice.impl.InvocationBuilderImpl.invoke(InvocationBuilderImpl.java:59) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.MasterContext.invokeOnParticipant(MasterContext.java:261) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.MasterContext.invokeOnParticipants(MasterContext.java:245) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.MasterJobContext.lambda$tryStartJob$2(MasterJobContext.java:227) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.JobCoordinationService.lambda$submitToCoordinatorThread$46(JobCoordinationService.java:1024) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.JobCoordinationService.submitToCoordinatorThread(JobCoordinationService.java:1033) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.JobCoordinationService.submitToCoordinatorThread(JobCoordinationService.java:1023) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.MasterJobContext.tryStartJob(MasterJobContext.java:189) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.JobCoordinationService.tryStartJob(JobCoordinationService.java:933) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.JobCoordinationService.lambda$submitJob$1(JobCoordinationService.java:258) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.JobCoordinationService.lambda$submitToCoordinatorThread$46(JobCoordinationService.java:1024) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.JobCoordinationService.lambda$submitToCoordinatorThread$47(JobCoordinationService.java:1045) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.internal.util.executor.CompletableFutureTask.run(CompletableFutureTask.java:64) [hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.internal.util.executor.CachedExecutorServiceDelegate$Worker.run(CachedExecutorServiceDelegate.java:217) [hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source) [?:?]
	at java.lang.Thread.run(Unknown Source) [?:?]
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.executeRun(HazelcastManagedThread.java:64) [hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:80) [hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
Caused by: java.lang.Exception: Initialization script finished with non-zero exit code: 127
	at com.hazelcast.jet.python.PythonServiceContext.<init>(PythonServiceContext.java:101) ~[hazelcast-jet-python-4.4-SNAPSHOT.jar!/:4.4-SNAPSHOT]
	... 40 more
11:53:57.000 [hz.competent_lalande.cached.thread-3] ERROR com.hazelcast.jet.impl.MasterJobContext - Execution of job 'MLChurnDetector@2020-10-24T11:53:52', execution 052a-dc2a-e3c0-0001 failed
	Start time: 2020-10-24T11:53:53.785
	Duration: 3,214 ms
	To see additional job metrics enable JobConfig.storeMetricsAfterJobCompletion
 com.hazelcast.jet.JetException: PythonService initialization failed: java.lang.Exception: Initialization script finished with non-zero exit code: 127
	at com.hazelcast.jet.python.PythonServiceContext.<init>(PythonServiceContext.java:110) ~[hazelcast-jet-python-4.4-SNAPSHOT.jar!/:4.4-SNAPSHOT]
	at com.hazelcast.jet.python.PythonService.createContextWithRetry(PythonService.java:111) ~[hazelcast-jet-python-4.4-SNAPSHOT.jar!/:4.4-SNAPSHOT]
	at com.hazelcast.jet.python.PythonService.lambda$factory$9a0f8f23$1(PythonService.java:91) ~[hazelcast-jet-python-4.4-SNAPSHOT.jar!/:4.4-SNAPSHOT]
	at com.hazelcast.function.FunctionEx.apply(FunctionEx.java:47) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.processor.ProcessorSupplierWithService.init(ProcessorSupplierWithService.java:55) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.execution.init.ExecutionPlan.initProcSuppliers(ExecutionPlan.java:327) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.execution.init.ExecutionPlan.initialize(ExecutionPlan.java:158) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.execution.ExecutionContext.initialize(ExecutionContext.java:138) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.JobExecutionService.lambda$initExecution$808bb183$1(JobExecutionService.java:240) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.function.RunnableEx.run(RunnableEx.java:30) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.util.Util.doWithClassLoader(Util.java:446) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.JobExecutionService.initExecution(JobExecutionService.java:240) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.operation.InitExecutionOperation.run(InitExecutionOperation.java:73) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.spi.impl.operationservice.Operation.call(Operation.java:184) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.call(OperationRunnerImpl.java:227) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:216) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationExecutorImpl.run(OperationExecutorImpl.java:406) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationExecutorImpl.runOrExecute(OperationExecutorImpl.java:433) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.doInvokeLocal(Invocation.java:596) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.doInvoke(Invocation.java:581) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.invoke0(Invocation.java:540) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.invoke(Invocation.java:237) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.spi.impl.operationservice.impl.InvocationBuilderImpl.invoke(InvocationBuilderImpl.java:59) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.MasterContext.invokeOnParticipant(MasterContext.java:261) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.MasterContext.invokeOnParticipants(MasterContext.java:245) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.MasterJobContext.lambda$tryStartJob$2(MasterJobContext.java:227) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.JobCoordinationService.lambda$submitToCoordinatorThread$46(JobCoordinationService.java:1024) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.JobCoordinationService.submitToCoordinatorThread(JobCoordinationService.java:1033) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.JobCoordinationService.submitToCoordinatorThread(JobCoordinationService.java:1023) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.MasterJobContext.tryStartJob(MasterJobContext.java:189) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.JobCoordinationService.tryStartJob(JobCoordinationService.java:933) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.JobCoordinationService.lambda$submitJob$1(JobCoordinationService.java:258) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.JobCoordinationService.lambda$submitToCoordinatorThread$46(JobCoordinationService.java:1024) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.jet.impl.JobCoordinationService.lambda$submitToCoordinatorThread$47(JobCoordinationService.java:1045) ~[hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.internal.util.executor.CompletableFutureTask.run(CompletableFutureTask.java:64) [hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.internal.util.executor.CachedExecutorServiceDelegate$Worker.run(CachedExecutorServiceDelegate.java:217) [hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source) [?:?]
	at java.lang.Thread.run(Unknown Source) [?:?]
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.executeRun(HazelcastManagedThread.java:64) [hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:80) [hazelcast-jet-enterprise-4.4-SNAPSHOT.jar!/:?]
Caused by: java.lang.Exception: Initialization script finished with non-zero exit code: 127
	at com.hazelcast.jet.python.PythonServiceContext.<init>(PythonServiceContext.java:101) ~[hazelcast-jet-python-4.4-SNAPSHOT.jar!/:4.4-SNAPSHOT]
	... 40 more
```

Occurs at least on 4.3 and 4.4-SNAPSHOT from 23/10/2020

--- a/extensions/python/src/main/java/com/hazelcast/jet/python/PythonServiceContext.java
+++ b/extensions/python/src/main/java/com/hazelcast/jet/python/PythonServiceContext.java
@@ -18,6 +18,7 @@ package com.hazelcast.jet.python;
 import com.hazelcast.internal.nio.IOUtil;
 import com.hazelcast.jet.JetException;
 import com.hazelcast.jet.core.ProcessorSupplier;
+import com.hazelcast.jet.impl.util.ExceptionUtil;
 import com.hazelcast.jet.impl.util.Util;
 import com.hazelcast.logging.ILogger;
 
@@ -76,6 +77,7 @@ class PythonServiceContext {
     PythonServiceContext(ProcessorSupplier.Context context, PythonServiceConfig cfg) {
         logger = context.jetInstance().getHazelcastInstance().getLoggingService()
                         .getLogger(getClass().getPackage().getName());
+        checkIfPythonIsAvailable();
         try {
             long start = System.nanoTime();
             runtimeBaseDir = recreateRuntimeBaseDir(context, cfg);
@@ -105,6 +107,14 @@ class PythonServiceContext {
                     TimeUnit.NANOSECONDS.toMillis(System.nanoTime() - start)));
         } catch (Exception e) {
             throw new JetException("PythonService initialization failed: " + e, e);
+        }
+    }
+
+    private void checkIfPythonIsAvailable() {
+        try {
+            new ProcessBuilder("python3", "--version").start().waitFor();
+        } catch (Exception e) {
+            throw ExceptionUtil.sneakyThrow(new Exception("python3 is not available", e));
         }
     }
 

--- a/extensions/python/src/main/java/com/hazelcast/jet/python/PythonServiceContext.java
+++ b/extensions/python/src/main/java/com/hazelcast/jet/python/PythonServiceContext.java
@@ -29,7 +29,6 @@ import java.io.InputStream;
 import java.io.InputStreamReader;
 import java.io.OutputStream;
 import java.io.PrintWriter;
-import java.nio.charset.StandardCharsets;
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.nio.file.attribute.PosixFilePermission;

--- a/extensions/python/src/main/java/com/hazelcast/jet/python/PythonServiceContext.java
+++ b/extensions/python/src/main/java/com/hazelcast/jet/python/PythonServiceContext.java
@@ -29,6 +29,7 @@ import java.io.InputStream;
 import java.io.InputStreamReader;
 import java.io.OutputStream;
 import java.io.PrintWriter;
+import java.nio.charset.StandardCharsets;
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.nio.file.attribute.PosixFilePermission;
@@ -114,14 +115,16 @@ class PythonServiceContext {
         try {
             Process process = new ProcessBuilder("python3", "--version").redirectErrorStream(true).start();
             process.waitFor();
-            String output = new String(readFully(process.getInputStream()));
-            if (process.exitValue() != 0) {
-                logger.severe("python3 version check returned non-zero exit value, output: " + output);
-                throw new IllegalStateException("python3 is not available");
-            }
-            if (!output.startsWith("Python 3")) {
-                logger.severe("python3 version check returned unknown version, output: " + output);
-                throw new IllegalStateException("python3 is not available");
+            try (InputStream inputStream = process.getInputStream()) {
+                String output = new String(readFully(inputStream), UTF_8);
+                if (process.exitValue() != 0) {
+                    logger.severe("python3 version check returned non-zero exit value, output: " + output);
+                    throw new IllegalStateException("python3 is not available");
+                }
+                if (!output.startsWith("Python 3")) {
+                    logger.severe("python3 version check returned unknown version, output: " + output);
+                    throw new IllegalStateException("python3 is not available");
+                }
             }
         } catch (Exception e) {
             throw new IllegalStateException("python3 is not available", e);

--- a/extensions/python/src/main/java/com/hazelcast/jet/python/PythonServiceContext.java
+++ b/extensions/python/src/main/java/com/hazelcast/jet/python/PythonServiceContext.java
@@ -18,7 +18,6 @@ package com.hazelcast.jet.python;
 import com.hazelcast.internal.nio.IOUtil;
 import com.hazelcast.jet.JetException;
 import com.hazelcast.jet.core.ProcessorSupplier;
-import com.hazelcast.jet.impl.util.ExceptionUtil;
 import com.hazelcast.jet.impl.util.Util;
 import com.hazelcast.logging.ILogger;
 
@@ -114,7 +113,7 @@ class PythonServiceContext {
         try {
             new ProcessBuilder("python3", "--version").start().waitFor();
         } catch (Exception e) {
-            throw ExceptionUtil.sneakyThrow(new Exception("python3 is not available", e));
+            throw new IllegalStateException("python3 is not available", e);
         }
     }
 

--- a/extensions/python/src/main/java/com/hazelcast/jet/python/PythonServiceContext.java
+++ b/extensions/python/src/main/java/com/hazelcast/jet/python/PythonServiceContext.java
@@ -38,6 +38,7 @@ import java.util.Objects;
 import java.util.concurrent.TimeUnit;
 
 import static com.hazelcast.jet.impl.util.IOUtil.copyStream;
+import static com.hazelcast.jet.impl.util.IOUtil.readFully;
 import static com.hazelcast.jet.impl.util.Util.editPermissionsRecursively;
 import static java.nio.charset.StandardCharsets.UTF_8;
 import static java.nio.file.attribute.PosixFilePermission.GROUP_WRITE;
@@ -75,7 +76,7 @@ class PythonServiceContext {
 
     PythonServiceContext(ProcessorSupplier.Context context, PythonServiceConfig cfg) {
         logger = context.jetInstance().getHazelcastInstance().getLoggingService()
-                        .getLogger(getClass().getPackage().getName());
+                .getLogger(getClass().getPackage().getName());
         checkIfPythonIsAvailable();
         try {
             long start = System.nanoTime();
@@ -111,7 +112,17 @@ class PythonServiceContext {
 
     private void checkIfPythonIsAvailable() {
         try {
-            new ProcessBuilder("python3", "--version").start().waitFor();
+            Process process = new ProcessBuilder("python3", "--version").redirectErrorStream(true).start();
+            process.waitFor();
+            String output = new String(readFully(process.getInputStream()));
+            if (process.exitValue() != 0) {
+                logger.severe("python3 version check returned non-zero exit value, output: " + output);
+                throw new IllegalStateException("python3 is not available");
+            }
+            if (!output.startsWith("Python 3")) {
+                logger.severe("python3 version check returned unknown version, output: " + output);
+                throw new IllegalStateException("python3 is not available");
+            }
         } catch (Exception e) {
             throw new IllegalStateException("python3 is not available", e);
         }

